src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -94,11 +84,13 @@ def __init__(
         should_index_tag_values: bool,
         is_output_sliced: bool,
         input_codec: Optional[Codec[Any]],
+        tags_validator: Callable[[Mapping[str, str]], bool],
```

Comment:
Is there any chance we can consider keeping the tags_validator in side of [MetricsIngestConfiguration](https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/configuration.py#L65)?
I feel like doing so will
- Allow us to have a single entity that describes the behaviour of the entire pipeline
- Make it easier to track and review future changes

Reply:
It seems like if we do that then we won't be able to do [this](https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/configuration.py#L1-L5). I also tried out locally your suggestion just to try it out but it leads to import sequence problems which suggests that it is probably not a good idea.

Reply:
Ooof that sucks, and that's definitely a code smell. I think I know what I'm going to do during project week :)
I guess we will pass it into for now

Reply:
Speaking of, it looks like with this refactor, the batcher still does not know which metric pipeline it is on, should we consider passing in a UseCaseKey (aka metric path key) as well for the killswitch and the ugliness at 411?
```python
if self.__should_index_tag_values:
    new_payload_v1
else:
    new_payload_v2
```
or do you prefer those 2 be refactored in the similar pattern as what this pr is doing with tag validation?

Reply:
My current thinking is that we would ideally inject the differences in the metric pipelines using dependency injection from a layer above this similar to this PR. `IndexerBatch` does not need to have any internal knowledge of UseCaseKey.